### PR TITLE
Show current session name in Launcher and web interface

### PR DESCRIPTION
**Pull request by rdumusc**
+ other cleanups and refactorings

### CODE REVIEW

File: tide/master/MasterApplication.cpp [hunk 1]
```diff
@@ -548,6 +561,7 @@ bool MasterApplication::notify(QObject* receiver, QEvent* event)
     {
         _handle(static_cast<const QTouchEvent*>(event));
     }
+    break;
```
Comment (tribal-tec) [inline tide/master/MasterApplication.cpp]: this looks odd. move it to the scope or get rid of `default`?

File: tide/master/rest/RestInterface.cpp [hunk 2]
```diff
@@ -137,8 +138,8 @@ class RestInterface::Impl
 };
 
 RestInterface::RestInterface(const uint16_t port, OptionsPtr options,
-                             Scene& scene, Configuration& config)
-    : _impl(new Impl(port, options, scene, config, false))
+                             Session& session, Configuration& config)
+    : _impl(new Impl(port, options, session, config, false))
```
Comment (tribal-tec) [inline tide/master/rest/RestInterface.cpp]: keep just passing `scene`?

File: tide/master/session/ContentValidator.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+#include "utils/log.h"
+
+#include <QtConcurrent>
+
+namespace
+{
+bool _canBeRestored(const ContentType type)
+{
+    // PixelStreams are external applications and can't be restored.
+    if (type == ContentType::pixel_stream)
```
Comment (tribal-tec) [inline tide/master/session/ContentValidator.cpp]: just `return type != ContentType::pixel_stream`?

File: tide/master/rest/RestInterface.cpp (hunk 2 above)
Comment (rdumusc) [reply tide/master/rest/RestInterface.cpp]: done

File: tide/master/session/ContentValidator.cpp (hunk 3 above)
Comment (rdumusc) [reply tide/master/session/ContentValidator.cpp]: done

**Pull request merged**